src/App.tsx

```diff
@@ -42,54 +61,85 @@ export function getReorderedGoods(
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  }
+
+  sort = (sortType: SortType) => () => {
+    this.setState({ sortType });
+  };
+
+reversed = () => () => {
+  this.setState(prevState => ({ isReversed: !prevState.isReversed }));
+}
+
+  reset = (sortType: SortType) => () => {
+    this.setState({ sortType, isReversed: false });
+  }
+
+  render() {
+    const goodies = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={this.state.sortType === SortType.ALPHABET
+              ? 'button is-info'
+              : 'button is-info is-light'}
+            onClick={this.sort(SortType.ALPHABET)}
+          >
+            Sort alphabetically
+          </button>
+          <button
+            type="button"
+            className={this.state.sortType === SortType.LENGTH
+              ? 'button is-success'
+              : 'button is-success is-light'}
+            onClick={this.sort(SortType.LENGTH)}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={this.state.isReversed
+              ? 'button is-warning'
+              : 'button is-warning is-light'}
+            onClick={this.reversed()}
+          >
+            Reverse
+          </button>
+          {!this.state.isReversed && this.state.sortType === SortType.NONE
+            ? ''
+            : (
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={this.reset(SortType.NONE)}
```

Comment:
what you pass to onClick should be a function. Here you are passing the result of calling that function to onClick (this.reset(SortType.NONE)) is a function call, not a function).Example: {this.reset}, also if you want to pass args {() => this.reset(SortType.NONE)}.Check all your code for this